libc/string/string_errsig.c

```diff
@@ -48,6 +52,35 @@ static const unsigned int error_codes_len = sizeof(error_codes) / sizeof(error_c
 
 const unsigned int signal_codes_len = sizeof(signal_codes) / sizeof(signal_codes[0]);
 
+
+char *perrorToFile(const char *msg)
+{
+	int errnoBefore = errno;
+
+	FILE *file = fopen("error.txt", "w+");
+
+	TEST_ASSERT_NOT_NULL(file);
+	TEST_ASSERT_NOT_EQUAL(-1, dup2(fileno(file), STDERR_FILENO));
```

Comment:
I'm thinking if in case dup2 fails, then the file wouldn't be closed, and could this result in the rest of the test cases failing even though only one might, Am i correct ?

----

@damianloew 
After giving it some thought, I'm thinking how to approach this because at the moment, in most tests, if something fails in the middle of the test, we then don't do free(), fclose(), etc. 
Is it worth switching to something like:
```C
if (OPERATION(RESOURCE) == NULL) {
    free(RESOURCE);
    TEST_FAIL_MESSAGE("OPERATION FAILED");
}
```
Or to stay with our current approach

Reply:
Too late, already done.